src/components/Images.js

```diff
@@ -0,0 +1,196 @@
+import React, { useState, useEffect } from 'react';
+
+import img1 from "../img/1.png";
+import img2 from "../img/2.png";
+import img3 from "../img/3.png";
+import img4 from "../img/4.png";
+import img5 from "../img/5.png";
+import img6 from "../img/6.png";
+
+const defaultImg = 'https://via.placeholder.com/261x261?text=CLICK'
+
+function Images() {
+    const imgs = [
+        {
+            name: 1,
+            loc: img1,
+            flipped: true,
+            completed: true
+        },
+        {
+            name: 2,
+            loc: img2,
+            flipped: true,
+            completed: true
+        },
+        {
+            name: 3,
+            loc: img3,
+            flipped: true,
+            completed: true
+        },
+        {
+            name: 4,
+            loc: img4,
+            flipped: true,
+            completed: true
+        },
+        {
+            name: 5,
+            loc: img5,
+            flipped: true,
+            completed: true
+        },
+        {
+            name: 6,
+            loc: img6,
+            flipped: true,
+            completed: true
+        }
+    ]
+
+    const [finalGame, setfinalGame] = useState([])
+    const [preName, setPreName] = useState()
+    const [points, setPoints] = useState(0)
+    const [selectedImg, setSelectedImg] = useState()
+    const [hideStart, setHideStart] = useState()
+    const [playerName, setPlayerName] = useState()
+    const [timer, setTimer] = useState(3)
+    
+    useEffect(() => {
+        // const finalImages = [...imgs, ...imgs]
+        let finalImages = []
+        imgs.map(img => {
+            finalImages.push(JSON.parse(JSON.stringify(img)))
+            finalImages.push(JSON.parse(JSON.stringify(img)))
+        })
+        //2x Shuffling the images to make it more random
+        let shuffle = finalImages.sort(() => Math.random() - 0.5).map((img) => img)
+        let finalBoard = shuffle.sort(() => Math.random() - 0.5).map((img) => img)
```

Comment:
Any reason you have written this 2 times?

Reply:
In single shuffle 2 or more same images where adjacent to each other in 2 shuffle its just 1

Reply:
You should change logic, instead of using twice. Might be for more images you need to run it 3 or 4 times??
Shuffle logic should called once & it should do the job properly